### ISSUE TITLE
new feature: plot velocites in full vector form  

### ISSUE THREAD

**Issue by nathanieltarshish** [OceanPARCELS/parcels#96]
I wrote a new method that utilizes the `Basemap.quiver` module to produce vector plots of the velocity data. The usage of `Basemap` necessitates that `mpl_toolkits` be added to the requirements list.  The method, called `show_velocity()`, belongs to the `ParticleSet` class and relies on the ` temporal_interpolate_fullfield` method to find the velocities over the domain at any arbitrary time. It produces plots of the normalized velocity vector field colored by speed.

 `show_velocity()` has the following arguments: 
` t`: datetime or timedelta object that specifies time at which to plot velocity
`land`: boolean that is true iff land should be drawn on plot
 `latN, latS, lonW, lonE`: floats that delineate the geographic domain to be plotted. 

The user-specified domain is projected onto the velocity grid (using a new `nearest_index` function). This feature allows the user to be ignorant of the precise values of the velocity grid's lat/lon coordinates. If no domain is specified by the user, the method defaults to plotting the full domain. The method also plots the current locations of particles in the ParticleSet. Using the GlobCurrent example data set, after instantiating a grid and pset, calling `pset.show_velocity(t=datetime(2002, 1, 2), land=True)` produces the following sample output: 
![globcur_sample](https://cloud.githubusercontent.com/assets/20796154/17530408/62b955f8-5e45-11e6-86e5-9610a477fd98.png). 
By alternating between advecting the particles and saving velocity plots, one can create very useful animations: 
![animation_fast](https://cloud.githubusercontent.com/assets/20796154/17530534/f0f55e7a-5e45-11e6-9228-eef3d9de4c77.gif)

The new code is over on my fork along with a file called example_vector_plots.py that is located in the example folder. This file demonstrates the features of the new method with the MovingEddies and the GlobCurrent example datasets. If you are interested in including this feature, the master branch on my fork is ready to be merged with the main fork. I am looking forward to hearing any suggestions or feedback!

**Comment by StephenGriffies** [comment]
Very nice diagnostic.  It will be quite useful! Thanks, stephen

**Comment by mlange05** [comment]
This looks awesome! Can you please create a branch on the main repo for this and submit a Pull Request?

**Comment by erikvansebille** [comment]
Indeed, very cool!! It would be fantastic to land this into master, yes. Really cool eye-candy and a great showcase for why python trumps Fortran :-)

A few small comments:
1) How did you make the animation? Was that during runtime? Or using the netcdf file at the end (via the plotparticles.py script)?
2) Have a look at the `ipython_updates_clean` branch to see how `pset.show` can be made to automatically show the current time of the `pset`, so that users don't have to explicitly specify (in `particle.py` of ea79cc7). 
3) The `Basemap` is a pretty hefty dependence, and may not be available on all machines, so probably best to treat it just like matplotlib is treated, within the `try`/`except` clause on the top of `particle.py`
4) Should there be an argument to specify the max velocity, so that the colour bar doesn't change during the animation?

**Comment by nathanieltarshish** [comment]
Thanks for the feedback! I'm glad to see that you are interested in this feature. I'll make a branch and submit a pull request on your fork after I incorporate the improvements Erik brought up. In reply to your comments, Eric, 

1) I created the images, but not the animation, during runtime. I broke the advection into 6 hour intervals and called the `show_velocity` method between intervals. I also configured `show_velocity`  to save and number each plot rather than show it. I then constructed an animation from the images with the easy to use commandline tool, [ImageMagick](http://www.imagemagick.org/script/index.php), to create a .gif with the command `convert -delay 35  source*.jpg animated.gif`
2) I'll check out #97 and try to incorporate that as the default for  `show_velocity` 
3) That makes sense, will do. 
4) I'll make that an optional argument. 

Thanks for your feedback!

**Comment by nathanieltarshish** [comment]
I incorporated the changes discussed above and submitted them as pull request #98. In particular, there is now an optional argument `vmax` that sets the maximum velocity value of the color bar (so that it does not fluctuate in animations).  The default value for `t` is now also specified by `pset.particles[0].time`. I also changed the default `cmap` to `gist_ncar' which is shown below: 
![output0123](https://cloud.githubusercontent.com/assets/20796154/17573537/1449c92c-5f29-11e6-8ce5-baa7b72cffac.png)
